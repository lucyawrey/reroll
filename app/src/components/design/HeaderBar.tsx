import React from "react";
import { Navbar, Button, InputGroup, Form } from "react-bootstrap";
import { MdSearch } from "react-icons/md";
import Link from "next/link";

/**
 * A standard Header Bar used on every page
 */
function HeaderBar(): JSX.Element {
  return (
    <Navbar variant="dark" bg="dark" expand="lg" className="justify-content-between">
      <Link href="/" passHref>
        <Navbar.Brand>Reroll</Navbar.Brand>
      </Link>
      <InputGroup className="animate">
        <Form.Control type="text" placeholder="Search everything..." />
        <InputGroup.Append>
          <Button variant="secondary"><MdSearch /></Button>
        </InputGroup.Append>
      </InputGroup>
    </Navbar>
  );
}

export default HeaderBar;