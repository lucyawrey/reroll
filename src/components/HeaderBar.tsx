import { Theme } from "@material-ui/core";
import AppBar from "@material-ui/core/AppBar";
import Badge from "@material-ui/core/Badge";
import IconButton from "@material-ui/core/IconButton";
import InputBase from "@material-ui/core/InputBase";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import { fade } from "@material-ui/core/styles/colorManipulator";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MailIcon from "@material-ui/icons/Mail";
import MenuIcon from "@material-ui/icons/Menu";
import MoreIcon from "@material-ui/icons/MoreVert";
import NotificationsIcon from "@material-ui/icons/Notifications";
import SearchIcon from "@material-ui/icons/Search";
import { makeStyles } from "@material-ui/styles";
import Link from "next/link";
import React, { MouseEvent } from "react";
import Router from "next/router";

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    width: "100%",
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block",
    },
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginRight: theme.spacing(2),
    marginLeft: 0,
    width: "100%",
    paddingRight: theme.spacing(),
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing(3),
      width: "auto",
    },
  },
  searchIcon: {
    width: theme.spacing(9),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  inputRoot: {
    color: "inherit",
    width: "100%",
  },
  inputInput: {
    paddingTop: theme.spacing(),
    paddingRight: theme.spacing(),
    paddingBottom: theme.spacing(),
    paddingLeft: theme.spacing(10),
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("md")]: {
      width: 200,
    },
  },
  sectionDesktop: {
    display: "none",
    [theme.breakpoints.up("md")]: {
      display: "flex",
    },
  },
  sectionMobile: {
    display: "flex",
    [theme.breakpoints.up("md")]: {
      display: "none",
    },
  },
}));

/**
 * The static header bar at the top of the app. This code was written by the MaterialUI team
 * and the only work we did was getting it to play nice with TypeScript.
 */
function HeaderBar(props: any) {
  const classes = useStyles(props);
  const [anchorEl, setAnchorEl] = React.useState(null);
  const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState(null);

  const isMenuOpen = Boolean(anchorEl);
  const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);

  function handleProfileMenuOpen(event: MouseEvent<any>) {
    setAnchorEl(event.currentTarget);
  }

  function handleMobileMenuClose() {
    setMobileMoreAnchorEl(null);
  }

  function handleMenuClose() {
    setAnchorEl(null);
    handleMobileMenuClose();
  }

  function handleMobileMenuOpen(event: MouseEvent<any>) {
    setMobileMoreAnchorEl(event.currentTarget);
  }

  const renderMenu = (
    <Menu
      anchorEl={anchorEl}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={isMenuOpen}
      onClose={handleMenuClose}
    >
      <MenuItem onClick={() => {Router.push("/profile")}}>Profile</MenuItem>

      <Link href="/settings" passHref>
        <MenuItem onClick={handleMenuClose}>Settings</MenuItem>
      </Link>
      <Link href="/logout" passHref>
        <MenuItem onClick={handleMenuClose}>Log Out</MenuItem>
      </Link>
    </Menu>
  );

  const renderMobileMenu = (
    <Menu
      anchorEl={mobileMoreAnchorEl}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={isMobileMenuOpen}
      onClose={handleMobileMenuClose}
    >
      <MenuItem>
        <IconButton color="inherit">
          <Badge badgeContent={4} color="secondary">
            <MailIcon />
          </Badge>
        </IconButton>
        <p>Messages</p>
      </MenuItem>
      <MenuItem>
        <IconButton color="inherit">
          <Badge badgeContent={11} color="secondary">
            <NotificationsIcon />
          </Badge>
        </IconButton>
        <p>Notifications</p>
      </MenuItem>
      <MenuItem onClick={handleProfileMenuOpen}>
        <IconButton color="inherit">
          <AccountCircle />
        </IconButton>
        <p>Account</p>
      </MenuItem>
    </Menu>
  );

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton className={classes.menuButton} color="inherit" aria-label="Open drawer">
            <MenuIcon />
          </IconButton>
          <Typography className={classes.title} variant="h6" color="inherit" noWrap>
            Reroll
          </Typography>
          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Search Everythingâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput,
              }}
            />
          </div>
          <div className={classes.grow} />
          <div className={classes.sectionDesktop}>
            <IconButton color="inherit">
              <Badge badgeContent={4} color="secondary">
                <MailIcon />
              </Badge>
            </IconButton>
            <IconButton color="inherit">
              <Badge badgeContent={17} color="secondary">
                <NotificationsIcon />
              </Badge>
            </IconButton>
            <IconButton
              aria-owns={isMenuOpen ? "material-appbar" : undefined}
              aria-haspopup="true"
              onClick={handleProfileMenuOpen}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
          </div>
          <div className={classes.sectionMobile}>
            <IconButton aria-haspopup="true" onClick={handleMobileMenuOpen} color="inherit">
              <MoreIcon />
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
      {renderMenu}
      {renderMobileMenu}
    </div>
  );
}

export default HeaderBar;
