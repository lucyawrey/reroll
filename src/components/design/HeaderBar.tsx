import React from "react";
import { Navbar, Button, InputGroup, Form } from "react-bootstrap";
import { MdSearch } from "react-icons/md";
import Link from "next/link";

function HeaderBar() {
  return (
    <Navbar variant="dark" bg="dark" expand="lg" className="justify-content-between">
      <Link href="/" passHref>
        <Navbar.Brand>Reroll</Navbar.Brand>
      </Link>
      <InputGroup className="animate">
        <Form.Control type="text" placeholder="Search everything..." />
        <InputGroup.Append>
          <Button variant="secondary"><MdSearch /></Button>
        </InputGroup.Append>
      </InputGroup>
    </Navbar>
  );
}

export default HeaderBar;