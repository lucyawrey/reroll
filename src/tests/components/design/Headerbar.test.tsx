import React from "react";
import renderer from "react-test-renderer";
import HeaderBar from "../../../components/design/HeaderBar";

describe('Render Headerbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HeaderBar/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

