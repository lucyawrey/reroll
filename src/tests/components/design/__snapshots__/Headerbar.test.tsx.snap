// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Headerbar renders correctly 1`] = `
<nav
  className="justify-content-between navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Reroll
  </a>
  <div
    className="animate input-group"
  >
    <input
      className="form-control"
      placeholder="Search everything..."
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </button>
    </div>
  </div>
</nav>
`;
